cmd/syncthing/gui_auth.go

```diff
@@ -128,6 +116,60 @@ func basicAuthAndSessionMiddleware(cookieName string, cfg config.GUIConfiguratio
 	})
 }
 
+func auth(username string, password string, cfg config.GUIConfiguration) bool {
+	if cfg.AuthMode == config.AuthModeLDAP {
+		return authLDAP(username, password, cfg.LDAPAuth)
+	} else {
+		return authStatic(username, password, cfg.User, cfg.Password)
+	}
+}
+
+func authStatic(username string, password string, configUser string, configPassword string) bool {
+
+	if username != configUser {
+		return false
+	}
+
+	configPasswordBytes := []byte(configPassword)
+	passwordBytes := []byte(password)
+	return bcrypt.CompareHashAndPassword(configPasswordBytes, passwordBytes) == nil
```

Comment:
This might have been this way, but this just hit me. 

Perhaps move the username check after the hash check to prevent timing attacks on username probing.

Reply:
Currently both username and password is checked, and failure on either results in a short random delay before returning an error.